tests/test_zipcode.py

```diff
@@ -0,0 +1,82 @@
+# tests.test_zipcode
+# Test the zipcode to latitutide,longitude lookup database.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Mon May 08 12:35:21 2017 -0400
+#
+# Copyright (C) 2016 Bengfort.com
+# For license information, see LICENSE.txt
+#
+# ID: test_zipcode.py [] benjamin@bengfort.com $
+
+"""
+Test the zipcode to latitutide,longitude lookup database.
+"""
+
+##########################################################################
+## Imports and Fixtures
+##########################################################################
+
+import os
+import unittest
+
+from hanish.zipcode import ZipCodeDB
+from hanish.exceptions import HanishValueError
+
+
+FIXTURES = os.path.join(os.path.dirname(__file__), "..", "fixtures")
+ZIPCODES = os.path.join(FIXTURES, "ziplatlon.csv")
+
+
+##########################################################################
+## ZipCodeDB Tests
+##########################################################################
+
+class ZipCodeDBTests(unittest.TestCase):
+
+    def test_database_loading_lookup_and_close(self):
```

Comment:
It absolutely could be two separate tests, and there are a number of places that I combined two-three tests into one to get maximum coverage, primarily for development speed (4 hour time limit and all).  My thought was this, the separate tests would look something like this:

```python
def test_database_loading(self):
    zipdb = ZipCodeDB.load(ZIPCODES)
    assert zipped is loaded 

def test_database_lookups(self):
    zipdb = ZipCodeDB.load(ZIPCODES)
    assert zipdb is loaded 
    # perform lookups 

def test_database_not_found(self):
    zipdb = ZipCodeDB.load(ZIPCODES)
    assert zipdb is loaded 
    # perform not found lookups 

def test_database_close(self):
    zipdb = ZipCodeDB.load(ZIPCODES)
    assert zipdb is loaded 
    zipdb.close() 
    assert zipdb is closed 
```

So, clearly some repetitive code. The way to handle this is to use `setUp` and `tearDown` functions to do the loading and the load assertions. However, because I can't control the order that tests are run using nosetest, to ensure this would work properly I would have to create two TestCase classes, etc. etc. 

So for the sake of time, I went for coverage rather than quantity, and as I mentioned in an earlier comment; I would iterate on the tests as we continued iterating on the project. 

Just a note (I know you guys use Ruby) -- this is a place where RSpec/BDD style testing has a _clear_ advantage over this style, because I could have used a context or something similar to isolate the load test from the rest of the behavior!